components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -695,6 +749,81 @@ public Response apisApiIdPut(String apiId, APIDTO body, String ifMatch, MessageC
         } catch (FaultGatewaysException e) {
             String errorMessage = "Error while updating API : " + apiId;
             RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        } catch (CryptoException e) {
+            String errorMessage = "Error while encrypting the secret key of API : " + apiId;
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        } catch (ParseException e) {
+            String errorMessage = "Error while parsing endpoint config of API : " + apiId;
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        }
+        return null;
+    }
+
+    // AWS Lambda: rest api operation to get ARNs
+    @Override
+    public Response apisApiIdAmznResourceNamesGet(String apiId, MessageContext messageContext) {
+        JSONObject arns = new JSONObject();
+        try {
+            String tenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            API api = apiProvider.getAPIbyUUID(apiId, tenantDomain);
+            String endpointConfigString = api.getEndpointConfig();
+            if (!StringUtils.isEmpty(endpointConfigString)) {
+                JSONParser jsonParser = new JSONParser();
+                JSONObject endpointConfig = (JSONObject) jsonParser.parse(endpointConfigString);
+                if (endpointConfig != null) {
+                    if (endpointConfig.containsKey(APIConstants.AMZN_ACCESS_KEY) &&
+                            endpointConfig.containsKey(APIConstants.AMZN_SECRET_KEY)) {
+                        String accessKey = (String) endpointConfig.get(APIConstants.AMZN_ACCESS_KEY);
+                        String secretKey = (String) endpointConfig.get(APIConstants.AMZN_SECRET_KEY);
+                        AWSCredentialsProvider credentialsProvider;
+                        if (StringUtils.isEmpty(accessKey) && StringUtils.isEmpty(secretKey)) {
+                            credentialsProvider = InstanceProfileCredentialsProvider.getInstance();
+                        } else if (!StringUtils.isEmpty(accessKey) && !StringUtils.isEmpty(secretKey)) {
+                            if (secretKey.length() == APIConstants.AWS_ENCRYPTED_SECRET_KEY_LENGTH) {
+                                CryptoUtil cryptoUtil = CryptoUtil.getDefaultCryptoUtil();
+                                secretKey = new String(cryptoUtil.base64DecodeAndDecrypt(secretKey),
+                                        APIConstants.DigestAuthConstants.CHARSET);
+                            }
+                            BasicAWSCredentials awsCredentials = new BasicAWSCredentials(accessKey, secretKey);
+                            credentialsProvider = new AWSStaticCredentialsProvider(awsCredentials);
+                        } else {
+                            throw new SdkClientException("Missing AWS Credentials");
```

Comment:
I think this should be an APIManagementException since we are throwing.

Reply:
logged error instead of throwing